json-rpc/src/methods.rs

```diff
@@ -502,6 +495,33 @@ async fn get_account_transactions(
     Ok(all_txs)
 }
 
+//
+// Proofs API
+//
+
+/// Returns transactions by range with proofs
+async fn get_transactions_with_proofs(
+    service: JsonRpcService,
+    request: JsonRpcRequest,
+) -> Result<Option<TransactionListWithProof>> {
```

Comment:
you need return view, add a test to see what is json looks like

Reply:
need return onchain blob, instead of decoded Transaction data, imaging you are creating client, how you do the validation, you use blob to create hash and validate with proof, then decode the blob for the data.